File: HW/HIL_testing/protoip_project/ip_design/src/foo_user.cpp
/* 
* icl::protoip
* Author: asuardi <https://github.com/asuardi>
* Date: November - 2014
*/


#include "foo_data.h"
#include "user_fgm_mpc.h"


void foo_user(  data_t_x_hat_in x_hat_in_int[X_HAT_IN_LENGTH],
				data_t_u_opt_out u_opt_out_int[U_OPT_OUT_LENGTH])
{
	fgm_mpc(x_hat_in_int, u_opt_out_int);
}

File: HW/HIL_testing/protoip_project/ip_design/src/foo_user.m
%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% icl::protoip
% Author: asuardi <https://github.com/asuardi>
% Date: November - 2014
%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


function [u_opt_out_int] = foo_user(project_name,x_hat_in_int)


	% load project configuration parameters: input and output vectors (name, size, type, NUM_TEST, TYPE_TEST)
	load_configuration_parameters(project_name);

	%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
	% compute with Matlab and save in a file simulation results u_opt_out_int
	for i=1:U_OPT_OUT_LENGTH
		u_opt_out_int(i)=0;
		for i_x_hat = 1:X_HAT_IN_LENGTH
			u_opt_out_int(i)=u_opt_out_int(i) + x_hat_in_int(i_x_hat);
		end
	end

end

